src/EditorFeatures/Core/EditorConfigSettings/Extensions/SolutionExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions
+{
+    internal static class SolutionExtensions
+    {
+        public static ImmutableArray<Project> GetProjectsForPath(this Solution solution, string givenPath)
+        {
+            var givenFolder = new DirectoryInfo(givenPath).Parent;
```

Comment:
If `givenPath` were a path to a directory such as "/users/joey/source/roslyn/src/Compilers/", then `givenFolder` would point to "/users/joey/source/roslyn/src". I suggest using `Path.GetDirectoryName()` to strip off the filename as opposed to simply moving to the Parent.
```suggestion
            var givenFolder = new DirectoryInfo(Path.GetDirectoryName(givenPath));
```